app/controllers/pieces_controller.rb

```diff
@@ -4,22 +4,24 @@ def update
     @piece = Piece.find(params[:id])
     @game = Game.find(@piece.game_id)
 
-    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i)
+    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i) && !@piece.moving_into_check?(params[:y_coord].to_i, params[:x_coord].to_i) 
```

Comment:
I am able to recreate these same errors Edwin. Must be a problem with the moving_into_check...will look into it